### PR TITLE
Re-sync to main

### PR DESCRIPTION
**Pull request by 00JCIV00**
Sync all changes from main into personal fork in prep for a PR.

### CODE REVIEW

**Comment by 00JCIV00** [comment]
De-conflicted and good to go from my end. It may require a `zig fmt` run as I don't use that in my code.

File: build.zig
Comment (rockorager) [inline build.zig:1]: Let's change this back to `main.zig`. I originally thought I wanted to change the entry point but let's keep it as main for now.

File: src/Window.zig [hunk 1]
```diff
@@ -3,14 +3,14 @@ const ziglyph = @import("ziglyph");
 const WordIterator = ziglyph.WordIterator;
 const GraphemeIterator = ziglyph.GraphemeIterator;
 
-const Screen = @import("Screen.zig");
+pub const Screen = @import("Screen.zig");
```
Comment (rockorager) [inline src/Window.zig:6]: Let's drop this `pub`. I previously exported `Screen` via `main.zig` so this isn't needed

File: src/Window.zig [hunk 2]
```diff
@@ -3,14 +3,14 @@ const ziglyph = @import("ziglyph");
...
 const Cell = @import("Cell.zig");
 const Segment = @import("Cell.zig").Segment;
 const gw = @import("gwidth.zig");
 
 const log = std.log.scoped(.window);
 
-const Window = @This();
+pub const Window = @This();
```
Comment (rockorager) [inline src/Window.zig:13]: We don't need a `pub` here since this is exported whenever the file is `@import`ed

File: src/vaxis.zig [hunk 3]
```diff
@@ -2,23 +2,36 @@ const std = @import("std");
...
 
+
+/// Initialize a Vaxis application.
+pub fn init(comptime Event: type, opts: Options) !Vaxis(Event) {
+    return Vaxis(Event).init(opts);
+}
+test {
+    std.testing.refAllDecls(@This());
+}
+
```
Comment (rockorager) [inline src/vaxis.zig:34]: Let's keep this one in `main.zig`

File: src/vaxis.zig [hunk 4]
```diff
@@ -2,23 +2,36 @@ const std = @import("std");
...
-const Key = @import("Key.zig");
-const Screen = @import("Screen.zig");
-const InternalScreen = @import("InternalScreen.zig");
-const Window = @import("Window.zig");
-const Options = @import("Options.zig");
-const Style = @import("Cell.zig").Style;
-const Hyperlink = @import("Cell.zig").Hyperlink;
-const gwidth = @import("gwidth.zig");
-const Shape = @import("Mouse.zig").Shape;
-const Image = @import("Image.zig");
```
Comment (rockorager) [inline src/vaxis.zig:19]: I think I would prefer to keep these all private at this level and only expose these via `main.zig`. If there are any missing that are needed, let's add them there (but only if actually needed so as to keep the API small)

File: src/widgets/Table.zig [hunk 5]
```diff
@@ -4,10 +4,10 @@ const heap = std.heap;
 const mem = std.mem;
 const meta = std.meta;
 
-const vaxis = @import("../main.zig");
+const vaxis = @import("../vaxis.zig");
```
Comment (rockorager) [inline src/widgets/Table.zig:7]: Change back to `main.zig`

**Review by rockorager — CHANGES_REQUESTED** [review]
Left all my comments inline! Thanks!

**Comment by 00JCIV00** [comment]
No issues with the review on my part. All changes resolved.

**Comment by rockorager** [comment]
Did you push your local branch back up? I'm not seeing any updates in Github.

**Comment by rockorager** [comment]
Thanks again for all the updates to this one!

**Pull request merged**